### ISSUE TITLE
datastore: datastore.Client.RunAggregationQuery panics with nil pointer.

### ISSUE THREAD

**Issue by Jose-Francisco-neto** [googleapis/google-cloud-go#7365]
**Client**

datastore.Client

**Environment**

GKE datastore

**Go Environment**

$ go version go1.18.10 linux/amd64
$ go env
```
GO111MODULE=""
GOARCH="amd64"
GOBIN=""
GOCACHE="/home/jose.neto@db1.com.br/.cache/go-build"
GOENV="/home/jose.neto@db1.com.br/.config/go/env"
GOEXE=""
GOEXPERIMENT=""
GOFLAGS=""
GOHOSTARCH="amd64"
GOHOSTOS="linux"
GOINSECURE=""
GOMODCACHE="/home/jose.neto@db1.com.br/go/pkg/mod"
GONOPROXY="github.com/avenuesec"
GONOSUMDB="github.com/avenuesec"
GOOS="linux"
GOPATH="/home/jose.neto@db1.com.br/go"
GOPRIVATE="github.com/avenuesec"
GOPROXY="https://proxy.golang.org,direct"
GOROOT="/home/jose.neto@db1.com.br/.local/go"
GOSUMDB="sum.golang.org"
GOTMPDIR=""
GOTOOLDIR="/home/jose.neto@db1.com.br/.local/go/pkg/tool/linux_amd64"
GOVCS=""
GOVERSION="go1.18.10"
GCCGO="gccgo"
GOAMD64="v1"
AR="ar"
CC="gcc"
CXX="g++"
CGO_ENABLED="1"
GOMOD="/home/jose.neto@db1.com.br/Projects/Avenue/datastore-filler/go.mod"
GOWORK=""
CGO_CFLAGS="-g -O2"
CGO_CPPFLAGS=""
CGO_CXXFLAGS="-g -O2"
CGO_FFLAGS="-g -O2"
CGO_LDFLAGS="-g -O2"
PKG_CONFIG="pkg-config"
GOGCCFLAGS="-fPIC -m64 -pthread -fmessage-length=0 -fdebug-prefix-map=/tmp/go-build2388863386=/tmp/go-build -gno-record-gcc-switches"
```
**Code**

e.g.
```go
package main

import (
	"context"
	"fmt"

	"cloud.google.com/go/datastore"
)

func main() {
ctx := context.Background()
	dsClient, err := datastore.NewClient(ctx, ProjectName)
	if err != nil {
		panic(err)
	}

	query := datastore.NewQuery(AccountMigratorModelKind).Namespace(Namespace)

	agregator := query.NewAggregationQuery().WithCount("count")

	result, err := dsClient.RunAggregationQuery(ctx, agregator) // it panics here
	if err != nil {
		panic(err)
	}

	fmt.Println(result["count"])
}
```

**Expected behavior**

An query result or at leas a err result ?

**Actual behavior**

It panics with "nil dereference".

**Screenshots**

![image](https://user-images.githubusercontent.com/93209278/216986781-9f09f04a-9548-433d-812e-becb72bb073b.png)

**Additional context**

cloud.google.com/go/datastore v1.10.0
google.golang.org/genproto v0.0.0-20230113154510-dbe35b8444a5

**Comment by telpirion** [comment]
@Jose-Francisco-neto Can you please confirm that `AccountMigratorModelKind` and `Namespace` in the code snippet you provided are not nil?

**Comment by telpirion** [comment]
@Jose-Francisco-neto  Additional: Can you please provide a full stack trace?

**Comment by Jose-Francisco-neto** [comment]
Repo reproducing it: https://github.com/jfrancisco-neto/datastore-bug

`tarting: /home/jose.neto/go/bin/dlv dap --listen=127.0.0.1:42307 --log-dest=3 from /home/jose.neto/Projects/Personal/datastore-bug
DAP server listening at: 127.0.0.1:42307
Type 'dlv help' for list of commands.
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0xc1a9f6]

goroutine 1 [running]:
cloud.google.com/go/datastore.(*datastoreClient).RunAggregationQuery(0xc000113410, {0xded0f8, 0xc0000400c8}, 0xc0003009a0, {0x0, 0x0, 0x0})
	<autogenerated>:1 +0x96
cloud.google.com/go/datastore.(*Client).RunAggregationQuery(0xc00011cb80, {0xded0f8, 0xc0000400c8}, 0xc00056ba60)
	/home/jose.neto/go/pkg/mod/cloud.google.com/go/datastore@v1.10.0/query.go:663 +0x5ee
main.queryCount({0xded0f8, 0xc0000400c8}, 0xc00011cb80, {0x7ffc6bf2ff6c, 0x9}, {0x7ffc6bf2ff59, 0xc})
	/home/jose.neto/Projects/Personal/datastore-bug/main.go:87 +0xe8
main.main()
	/home/jose.neto/Projects/Personal/datastore-bug/main.go:113 +0x251
Process 282153 has exited with status 2
Detaching`

**Comment by Jose-Francisco-neto** [comment]
Ow, and I run it with vscode using this kind of args:
![image](https://user-images.githubusercontent.com/93209278/217414388-101e6753-b910-4292-a577-cc94ad677fe3.png)

**Comment by DustinJSilk** [comment]
Hi @telpirion, Are you sure this is fixed?
I'm experiencing the same issue. Both of those nil checks are false when i run them and the original issue still persists.

This is all it takes for mine to not work:

```go
client, err := datastore.NewClient(context.Background(), "my-project")
if err != nil {
  log.Fatalf("failed to create datastore client: %v", err)
}

qa := datastore.NewQuery("User").NewAggregationQuery().WithCount("count")
_, err = client.RunAggregationQuery(context.Background(), qa)
```

This is my error:

```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x48 pc=0x16029b3]

goroutine 1 [running]:
cloud.google.com/go/datastore.(*datastoreClient).RunAggregationQuery(0xc0001d00f0?, {0x1c687d0?, 0xc00011e000?}, 0x104b556?, {0x0?, 0x1a80140?, 0x22d3801?})
        <autogenerated>:1 +0x33
cloud.google.com/go/datastore.(*Client).RunAggregationQuery(0xc0000a8a80, {0x1c687d0, 0xc00011e000}, 0xc0005bff50)
        /Users/.../go/pkg/mod/cloud.google.com/go/datastore@v1.10.0/query.go:663 +0x436
main.main()
```

**Comment by dmotylev** [comment]
I confirm it is not fixed in `cloud.google.com/go/datastore v1.11.0` and the sample from the report panics with `runtime error: invalid memory address or nil pointer dereference`

**Comment by DustinJSilk** [comment]
@telpirion can we please reopen this issue?

This is not working for me either in the latest release v1.11.0

**Comment by dasper** [comment]
This is still an open issue. Tried two different datastores and different collections/kinds and got the same panic

```
aq := datastore.NewQuery(kind).Namespace("main").NewAggregationQuery().WithCount("")
r, err := dev.RunAggregationQuery(ctx, aq)
```

`panic: runtime error: invalid memory address or nil pointer dereference` in `datastore@v1.11.0/query.go:789`

**Comment by jordiserrano** [comment]
I'm having the exact same issue with cloud.google.com/go/datastore v1.11.0

**Comment by DustinJSilk** [comment]
@jordiserrano  Theres another [open issue ](https://github.com/googleapis/google-cloud-go/issues/7800) and [pr](https://github.com/googleapis/google-cloud-go/pull/7803), though neither seem to be getting any traction from google so it might help to give them a thumbs up

**Comment by bhshkh** [comment]
The fix is available in datastore 1.12.0

**Comment by bhshkh** [comment]
The fix is available in datastore 1.12.0
